core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -481,12 +487,49 @@ public static void deleteCancerStudy(String cancerStudyStableId) throws DaoExcep
             JdbcUtil.closeAll(DaoCancerStudy.class, con, pstmt, rs);
         }
     }
+    
+    /**
+     * Calls deleteCancerStudyByCascade() if cancer study exists by stable id.
+     * @param cancerStudyStableId
+     * @throws DaoException 
+     */
+    public static void deleteCancerStudyByCascade(String cancerStudyStableId) throws DaoException {
+        CancerStudy study = getCancerStudyByStableId(cancerStudyStableId);
+        if (study != null) {
+            deleteCancerStudyByCascade(study.getInternalId());
+        }
+    }
 
     /**
+     * Deletes a cancer study by internal id from db with foreign key constraints enforced.
+     * @param internalCancerStudyId
+     * @throws DaoException 
+     */
+    static void deleteCancerStudyByCascade(int internalCancerStudyId) throws DaoException {
```

Comment:
this probably should have a visibility specifier (public or private). Public is probably good.